front/src/components/MapComponent.vue

```diff
@@ -1,52 +1,84 @@
 <script setup lang="ts">
 import mapboxgl from "mapbox-gl";
+import type * as Geojson from "geojson";
 import { onMounted, ref, watch } from "vue";
 import { useRoute, useRouter } from "vue-router";
 import { useMapPointsStore } from "@/stores/mapPoints";
 
-const router = useRouter();
-const route = useRoute();
-const MAPBOX_ACCESS_TOKEN = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN;
-mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+import type { Point } from "@/stores/mapPoints";
+export interface DirectionResponse {
+  waypoints: Waypoint[];
+  routes: Route[];
+  code: string;
+}
 
-interface MapProps {
-  center: [number, number];
-  maxBounds: [[number, number], [number, number]];
+export interface Route {
+  legs: Leg[];
+  weight_name: string;
+  geometry: Geometry;
+  weight: number;
+  distance: number;
+  duration: number;
 }
 
-const map = ref<mapboxgl.Map | null>(null);
+export interface Geometry {
+  coordinates: Array<number[]>;
+  type: string;
+}
+
+export interface Leg {
+  steps: any[];
+  weight: number;
+  distance: number;
+  summary: string;
+  duration: number;
+}
+
+export interface Waypoint {
+  location: number[];
+  name: string;
+}
 
-const mapProps: MapProps = {
-  center: [2.3522, 48.8566],
-  maxBounds: [
-    [-180, -85],
-    [180, 85],
-  ],
+const MAPBOX_ACCESS_TOKEN = import.meta.env.VITE_MAPBOX_ACCESS_TOKEN; // Access token from environment variable
+const MAP_STYLE = "mapbox://styles/mapbox/streets-v12";
+const mapProps: mapboxgl.MapboxOptions = {
+  // Define map properties
+  container: "mapContainer",
+  style: MAP_STYLE,
+  center: [-1.553621, 47.218372], // starting position [lng, lat]
+  zoom: 13, // starting zoom
 };
 
+const router = useRouter();
+const route = useRoute();
+
+mapboxgl.accessToken = MAPBOX_ACCESS_TOKEN;
+
+interface MapState {
+  map: mapboxgl.Map | null; // Interface for the map state
+}
+//  let map: ref<mapboxgl.Map | null>(null); // Reactive ref for the map instance
+const map = ref<MapState>({
+  map: null, // Initialize the map as null initially
+});
+
 const initializeMap = () => {
   if (map.value) {
-    map.value.remove();
+    map.value.map?.remove();
```

Comment:
Too many map word in same line...

Which is for mapbox and which for the map ???

Rename your var for better clarity